### ISSUE TITLE
Python async functions

### ISSUE THREAD

**Issue by LanderMoerkerke** [kkoomen/vim-doge#93]
**Describe the solution you'd like**

Only sync functions are supported in Python. It would be nice to also include async functions in the detection algorithm.

eg:
```python
async def test(a: int) -> bool:
    pass
```

**Additional context**

I tried to update the regex myself but the \m at the start throws me off since it is not valid regex. 

```
\m(async )?def\s\+\%([^(]\+\)\s*(\(.\{-}\))\%(\s*->\s*\(.\{-}\)\)\?\s*:
```

Edit:

I'm currently reading the contribution guidelines, shouldn't be hard to implement these. Also, I now see the use of \m :sweat_smile:

**Comment by kkoomen** [comment]
Your regex is in Perl style, not in Vim style. I'll have a look at it later.

**Comment by kkoomen** [comment]
This feature has been merged and released in [v2.8.0](https://github.com/kkoomen/vim-doge/tree/v2.8.0).

Feel free to submit any new issues if you experience any unwanted behavior in the future. Thanks for your contribution.
